### DISCUSSION TITLE
Invoking a service when an event occurs

### DISCUSSION THREAD

**Discussion by imeghri-sami** [statelyai/xstate#4817] [Q&A] [answered]
Is there a method to trigger services when an event occurs? From my understanding of the docs, services (actors in v5) are used to run side effects functions or essentially handle actions and their results confined within the context of a state. I want to call a service during a transition in response to an event, without introducing a new state and returning to the previous state, or creating a new actor machine only to execute the service.

```
state : {
...
active : {
UPLOAD : {
target: 'active',
invoke : {
src : ... # invoke my service here !
}
}
}
... 
}
```

**Comment by davidkpiano** [comment, marked as answer]
You can [spawn actors](https://stately.ai/docs/spawn) instead of invoking them based on an event.

📖 Docs: https://stately.ai/docs/spawn
